packages/terra-switch/src/Switch.jsx

```diff
@@ -0,0 +1,134 @@
+import React, { useRef, useCallback } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import styles from './Switch.module.scss';
+import { enableFocusStyles, disableFocusStyles } from './_SwitchUtils';
+
+const cx = classNames.bind(styles);
+
+// Translations Required
+const SWITCH_STATE = Object.freeze({
+  ON: 'On',
+  OFF: 'Off',
+});
+
+const propTypes = {
+  /**
+   * Id of the Switch button.
+   */
+  buttonId: PropTypes.string.isRequired,
+  /**
+   * Id of the Switch label.
+   */
+  labelId: PropTypes.string.isRequired,
+  /**
+   * Whether or not the Switch is checked ("ON").
+   */
+  checked: PropTypes.bool,
+  /**
+   * Whether or not the Switch is disabled.
+   */
+  disabled: PropTypes.bool,
+  /**
+   * The label of the Switch component.
+   */
+  label: PropTypes.node.isRequired,
+  /**
+   * Callback function when switch value changes from ON / OFF.
+   * Parameters: 1. switch value 2. event.
+   */
+  onChange: PropTypes.func,
+};
+
+const defaultProps = {
+  checked: false,
+  disabled: false,
+  onChange: undefined,
+};
+
+const Switch = (props) => {
+  const {
+    buttonId,
+    labelId,
+    checked,
+    disabled,
+    onChange,
+    label,
+    ...customProps
+  } = props;
+
+  const sliderButton = useRef();
+
+  const handleOnClick = useCallback(
+    (event) => {
+    // See https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Button#Clicking_and_focus
+    // Button on Firefox, Safari and IE running on OS X does not receive focus when clicked.
+    // This will set focus on the button when clicked.
+      sliderButton.current.focus();
+      if (onChange) {
+        onChange(!checked, event);
+      }
+    },
+    [checked, onChange],
+  );
+
+  const switchClassNames = cx([
+    'switch',
+  ]);
+
+  const trayClassNames = cx([
+    'tray',
+    { selected: checked },
+    { unselected: !checked },
+    { 'is-disabled': disabled },
+  ]);
+
+  const sliderClassNames = cx([
+    'slider',
+    { 'is-disabled': disabled },
+  ]);
+
+  const statusLabelText = checked ? SWITCH_STATE.ON : SWITCH_STATE.OFF;
+
+  let mainClasses = cx('switch-wrapper');
+  if (customProps.className) {
+    mainClasses = `${mainClasses} ${customProps.className}`;
+  }
```

Comment:
After referring a few terra-form components, I believe we do not need to check for the presence of className in customProps, instead it can be added as follows
```suggestion
  const mainClasses = cx([
    'switch-wrapper',
    customProps.className,
  ]);
```

Reply:
Check was added to prevent `undefined` from getting appended as classname to component when custom classnames were not provided.

Reply:
That is probably because `undefined` was being appended as a string.
But if you added it during binding (as suggested above), `undefined` does not get appended.

Reply:
Actually, this is the correct way to be doing it, we do not want to bind a custom classname.

Reply:
updated : 48fce2d9ca73773ac6f9891feaa7f2a56b36f864